ArduPlane/tailsitter.cpp

```diff
@@ -267,19 +275,29 @@ bool QuadPlane::tailsitter_transition_vtol_complete(void) const
 {
     if (plane.fly_inverted()) {
         // transition immediately
+        gcs().send_text(MAV_SEVERITY_INFO, "Transition VTOL done, inverted flight");
         return true;
     }
     // for vectored tailsitters at zero pilot throttle
-    if ((plane.quadplane.get_pilot_throttle() < .05f) && plane.quadplane._is_vectored) {
+    if (plane.control_mode->is_vtol_man_throttle() && (plane.quadplane.get_pilot_throttle() < .05f) && plane.quadplane._is_vectored) {
         // if we are not moving (hence on the ground?) or don't know
         // transition immediately to tilt motors up and prevent prop strikes
         if (ahrs.groundspeed() < 1.0f) {
+            gcs().send_text(MAV_SEVERITY_INFO, "Transition VTOL done, zero throttle");
             return true;
         }
     }
-    if (labs(plane.ahrs.pitch_sensor) > tailsitter.transition_angle*100 ||
-        labs(plane.ahrs.roll_sensor) > tailsitter.transition_angle*100 ||
-        AP_HAL::millis() - transition_start_ms > 2000) {
+    const float trans_angle = get_tailsitter_transition_angle_vtol();
+    if (labs(plane.ahrs.pitch_sensor) > trans_angle*100) {
+        gcs().send_text(MAV_SEVERITY_INFO, "Transition VTOL done");
+        return true;
+    }
+    if (labs(plane.ahrs.roll_sensor) >  4500) {
+        gcs().send_text(MAV_SEVERITY_INFO, "Transition VTOL done, roll error");
+        return true;
+    }
+    if (AP_HAL::millis() - transition_start_ms >  ((trans_angle-(transition_initial_pitch*0.01f))/tailsitter.transition_rate_fw)*1500) {
```

Comment:
I dont understand these timeouts.....if the initial pitch is level (0) then its the time versus the angle??? angle - (0/rate)*1500 = angle
I think the time needed to make the angle from the initail pitch at the given rate should be 1.5 multiplied for the time out
time =( (trans angle - initial pitch ) / rate) * 1.5 with whatever units conversion is needed in there, right? depending on direction of transition the sign before initial pitch would change, I think...